doc/lua_api.md

```diff
@@ -6579,8 +6581,10 @@ Environment access
       Difference between `"A*"` and `"A*_noprefetch"` is that
       `"A*"` will pre-calculate the cost-data, the other will calculate it
       on-the-fly
-* `core.spawn_tree (pos, {treedef})`
+* `core.spawn_tree (pos, {treedef}, [vmanip])`
     * spawns L-system tree at given `pos` with definition in `treedef` table
+    * if a `vmanip` parameter is passed the tree spawns inside the VoxelManip instead.
+      Note that inside the mapgen environment the mapgen vmanip will be used in any case.
```

Comment:
what? why?
I don't see this in the code.

Reply:
There are two `spawn_tree` functions, the normal one and one for the `mapgen environment` (internally called `emerge environment`).
See here `spawn_ltree` and `make_ltree`.

https://github.com/minetest/minetest/blob/b63e988bd60089b5f0595442b974886c17d5e1c9/src/script/lua_api/l_env.cpp#L1309

https://github.com/minetest/minetest/blob/b63e988bd60089b5f0595442b974886c17d5e1c9/src/script/lua_api/l_env.cpp#L1616

Reply:
ah, I see. but this restriction should not exist.